fuzz/target.c

```diff
@@ -23,17 +24,47 @@
 
 #include "../src/libfsm/minimise_test_oracle.h"
 
+/* for fsm_capture_dump */
+/* FIXME: should this be a public interface? */
+#include "../src/libfsm/capture.h"
+
+/* Buffer for sanitized fuzzer input */
+#define MAX_FUZZER_DATA (64 * 1024)
+static uint8_t data_buf[MAX_FUZZER_DATA + 1];
+
+/* Should fuzzer harness code be built that compares behavior
+ * with PCRE? (Obviously, this depends on PCRE.) */
+#ifndef CMP_PCRE
+#define CMP_PCRE 0
+#endif
+
+#if CMP_PCRE
+#define PCRE2_CODE_UNIT_WIDTH 8
+#include <pcre2.h>
+
+static int
+compare_with_pcre(const char *pattern, struct fsm *fsm);
+#endif
+
 /* 10 seconds */
 #define TIMEOUT_USEC (10ULL * 1000 * 1000)
 
+/* for TRACK_TIMES and EXPENSIVE_CHECKS */
+#include "../src/libfsm/internal.h"
+
 enum run_mode {
-	MODE_DEFAULT,
+	MODE_REGEX,
+	MODE_REGEX_SINGLE_ONLY,
+	MODE_REGEX_MULTI_ONLY,
+	MODE_IDEMPOTENT_DET_MIN,
```

Comment:
Can you add these new modes to ci.yml too please? where currently we have:
```
mode: [ m, p, d ]
```

Reply:
Updated in 239927c3.